shared/p2p/options.go

```diff
@@ -2,35 +2,82 @@ package p2p
 
 import (
 	"fmt"
+	"io/ioutil"
 	"net"
+	"os"
+
+	peer "github.com/libp2p/go-libp2p-peer"
 
 	"github.com/libp2p/go-libp2p"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	"github.com/prysmaticlabs/prysm/shared/iputils"
+
 	filter "github.com/libp2p/go-maddr-filter"
 	ma "github.com/multiformats/go-multiaddr"
-	"github.com/prysmaticlabs/prysm/shared/iputils"
 )
 
 // buildOptions for the libp2p host.
 // TODO(287): Expand on these options and provide the option configuration via flags.
-// Currently, this is a random port and a (seemingly) consistent private key
-// identity.
-func buildOptions(cfg *ServerConfig) []libp2p.Option {
+func buildOptions(cfg *ServerConfig) ([]libp2p.Option, error) {
```

Comment:
Lets not change the API here. If you can't read from disk, log some error and do not apply the identity option.

Reply:
OK this is fixed.